pkgs/applications/window-managers/phosh/default.nix

```diff
@@ -36,9 +37,15 @@ let
     sha256 = "0a4qh5pgyjki904qf7qmvqz2ksxb0p8xhgl2aixfbhixn0pw6saw";
   };
 
-  executable = writeText "phosh" ''
+  executable = writeShellScript "phosh" ''
```

Comment:
Actually, I would just get rid of this altogether.

Add one downstream patch for `--debug` (or maybe if it could be replaced by an environment variable, pass it to service), another one for the `XDG_DATA_DIRS` shenanigans, and finally one upstream patch that allows replacing the hardcoded FHS compositor path using a meson option (the bindir check will probably need to remain for development, though).